src/storage/IndexRepresentationStore.ts

```diff
@@ -41,7 +41,19 @@ export class IndexRepresentationStore extends PassthroughStore {
     if (isContainerIdentifier(identifier) && this.matchesPreferences(preferences)) {
       try {
         const indexIdentifier = { path: `${identifier.path}${this.indexName}` };
-        return await this.source.getRepresentation(indexIdentifier, preferences, conditions);
+        const index = await this.source.getRepresentation(indexIdentifier, preferences, conditions);
+        // We only care about the container metadata so preferences don't matter
+        const container = await this.source.getRepresentation(identifier, {}, conditions);
+        container.data.destroy();
+
+        // Uses the container metadata but with the index content-type.
+        // There is potential metadata loss if there is more representation-specific metadata,
+        // but that can be looked into once the issues above are resolved.
```

Comment:
which issues? link?

Reply:
Remark: metadata should (in general) be associated with the resource rather than its representation. Do we follow that principle in general?

Reply:
> 
> 
> which issues? link?

I meant the ones on top of the file, lines 22-26.

> 
> 
> Remark: metadata should (in general) be associated with the resource rather than its representation. Do we follow that principle in general?

The object is called `RepresentationMetadata` and the most commonly used field is `contentType`. So I would say no, the metadata is definitely not just associated with just the resource. Although contentType might be the only one. But should there be other ones the metadata object is the only place it can be put.

Now that we have graphs though, we could actually separate representation and resource graphs in the metadata and filter like that.

Reply:
All good.

> we could actually separate representation and resource graphs in the metadata

Excellent idea if needed in the future.